skcosmo/feature_selection/_greedy.py

```diff
@@ -0,0 +1,154 @@
+                X,
+                y,
+                accept_sparse="csc",
+                ensure_min_features=2,
+                force_all_finite=not tags.get("allow_nan", True),
+                multi_output=True,
+            )
+        else:
+            X = check_array(
+                X,
+                accept_sparse="csc",
+                ensure_min_features=2,
+                force_all_finite=not tags.get("allow_nan", True),
+            )
+
+        n_features = X.shape[1]
+
+        error_msg = (
+            "n_features_to_select must be either None, an "
+            "integer in [1, n_features - 1] "
+            "representing the absolute "
+            "number of features, or a float in (0, 1] "
+            "representing a percentage of features to "
+            f"select. Got {self.n_features_to_select}"
+        )
+        if self.n_features_to_select is None:
+            self.n_features_to_select_ = n_features // 2
+        elif isinstance(self.n_features_to_select, numbers.Integral):
+            if not 0 < self.n_features_to_select < n_features:
+                raise ValueError(error_msg)
+            self.n_features_to_select_ = self.n_features_to_select
+        elif isinstance(self.n_features_to_select, numbers.Real):
+            if not 0 < self.n_features_to_select <= 1:
+                raise ValueError(error_msg)
+            self.n_features_to_select_ = int(n_features * self.n_features_to_select)
+        else:
+            raise ValueError(error_msg)
+
+        # the current mask corresponds to the set of features:
+        if current_mask is None:
+            current_mask = np.zeros(shape=n_features, dtype=bool)
+
+        n_iterations = self.n_features_to_select_ - np.count_nonzero(current_mask)
+
+        for _ in range(n_iterations):
+            new_feature_idx = self._get_best_new_feature(
+                self.scoring, X, y, current_mask
+            )
+            current_mask[new_feature_idx] = True
+
+            self.selected_ = np.array([*self.selected_, new_feature_idx])
+            self.support_ = current_mask
+
+        return self
+
+    def _get_best_new_feature(self, scorer, X, y, current_mask):
+        # Return the best new feature to add to the current_mask, i.e. return
+        # the best new feature to add (resp. remove) when doing forward
+        # selection (resp. backward selection)
+        candidate_feature_indices = self._get_candidate_features(current_mask)
+        scores = np.zeros(X.shape[1])
+
+        for feature_idx in candidate_feature_indices:
+            scores[feature_idx] = scorer(X, y, feature_idx)
+
+        return np.argmax(scores)
+
+    def _get_support_mask(self):
+        check_is_fitted(self, ["support_"])
+        return self.support_
+
+    def _get_candidate_features(self, current_mask):
+        return np.where(current_mask is False)[0]
+
+    def _more_tags(self):
+        return {
+            "requires_y": False,
```

Comment:
I thought, that it is False by the default, isn't it?

Reply:
Correct -- this is mainly for pipeline integration.